tsconfig.json

```diff
@@ -1,37 +1,15 @@
 {
     "compilerOptions": {
-        "target": "es2020", // 어떤 버전으로 컴파일
-        "allowSyntheticDefaultImports": true, // default export가 없는 모듈에서 default imports를 허용
-        "experimentalDecorators": true, // decorator 실험적 허용
-        "emitDecoratorMetadata": true, // 데코레이터가 있는 선언에 대해 특정 타입의 메타 데이터를 내보내는 실험적인 지원
-        "skipLibCheck": true, // 정의 파일 타입 체크 여부
-        "moduleResolution": "node", // commonJS -> node 에서 동작
-        "module": "ES2020", // import 문법
-        "strict": true, // 타입 검사 엄격하게
-        "pretty": true, // error 메시지 예쁘게
```

Comment:
- remove `pretty`: default value is `true` [#ref](https://www.typescriptlang.org/tsconfig/#pretty)
- remove `allowJs`: now the repository doesn't have js at all
- move build specific options to `tsconfig.build.json`: to make it easy to reuse main one